block_chain.go

```diff
@@ -128,5 +128,20 @@ func (bc *Blockchain) GetLatestBlock() *Block {
 		}
 		return false
 	})
+	if len(longestPath) == 0 {
+		return bc.root.Block
+	}
 	return longestPath[len(longestPath)-1].Block
 }
+
+func (bc *Blockchain) GenerateNewBlock(transaction []Transaction) *Block {
+	latestBlock := bc.GetLatestBlock()
+	newBlock := &Block{
+		Index:        latestBlock.Index + 1,
+		Timestamp:    uint64(time.Now().Unix()),
+		Transactions: transaction,
+		PreviousHash: latestBlock.calculateHash(),
```

Comment:
since Hash is the array of const size it would be better to define it as `[32]bytes` rather than `[]bytes`. `[]bytes` suggest that blocks may have hashes of different length, why it should be known size